Cargo.toml

```diff
@@ -341,12 +341,12 @@ reth-optimism-rpc = { path = "crates/optimism/rpc" }
 reth-payload-builder = { path = "crates/payload/builder" }
 reth-payload-primitives = { path = "crates/payload/primitives" }
 reth-payload-validator = { path = "crates/payload/validator" }
-reth-primitives = { path = "crates/primitives" }
+reth-primitives = { path = "crates/primitives", default-features = false, features = ["std"] }
```

Comment:
this is a bit dangerous because this now disables "zstd-codec" which is problematic, right @joshieDo ?
so we'd need to enable this somewhere else, I'm thinking on node builder

Reply:
IIRC zstd doesn't work on 32-bit architectures btw, so that's why it's not enabled by default.